snapcraft/services/provider.py

```diff
@@ -20,19 +20,16 @@
 from craft_application import ProviderService
 from overrides import overrides
 
+from snapcraft.const import SNAPCRAFT_ENVIRONMENT_VARIABLES
+
 
 class Provider(ProviderService):
     """Snapcraft specialization of the Lifecycle Service."""
 
     @overrides
     def setup(self) -> None:
-        if build_info := os.getenv("SNAPCRAFT_BUILD_INFO"):
-            self.environment["SNAPCRAFT_BUILD_INFO"] = build_info
-        if image_info := os.getenv("SNAPCRAFT_IMAGE_INFO"):
-            self.environment["SNAPCRAFT_IMAGE_INFO"] = image_info
-        if experimental_extensions := os.getenv(
-            "SNAPCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS"
-        ):
-            self.environment["SNAPCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS"] = (
-                experimental_extensions
-            )
+        """Add snapcraft environment variables to the build environment."""
+        super().setup()
+        for variable in SNAPCRAFT_ENVIRONMENT_VARIABLES:
```

Comment:
Given https://github.com/canonical/craft-application/pull/416:

should we be doing this upstream in the `craft-application` ProviderService? E.g. for each item in the app's config, forward `{APPNAME}_{ITEM}`, and for each item in the craft-application config forward `CRAFT_{ITEM}`?

Reply:
Yes, I think that would fit nicely in craft-application's ProviderService.  

We can drop this in Snapcraft after that lands upstream.